aptos-move/e2e-move-tests/src/tests/remote_state.rs

```diff
@@ -282,3 +283,127 @@ async fn upgrade_package_via_object() {
     assert_eq!(val, 300);
 }
+
+#[ignore]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_option() {
+    let mut h = MoveHarness::new_with_remote_state(AptosBaseUrl::Testnet, 6691904943);
+    h.enable_features(
+        vec![
+            FeatureFlag::VM_BINARY_FORMAT_V8,
+            FeatureFlag::ENABLE_FUNCTION_VALUES,
+            FeatureFlag::ENABLE_ENUM_TYPES,
+        ],
+        vec![
+            FeatureFlag::ENABLE_ENUM_OPTION,
+            FeatureFlag::NEW_OPTION_MODULE,
+        ],
+    );
+
+    let existing_account_addr = AccountAddress::from_hex_literal(TESTNET_ACCOUNT_ADDR).unwrap();
+    let existing_account = h
+        .executor
+        .rotate_account_authentication_key(existing_account_addr);
+
+    let build_options = aptos_framework::BuildOptions {
+        bytecode_version: Some(8),
+        language_version: Some(LanguageVersion::V2_2),
+        ..Default::default()
+    };
+    let status: aptos_types::transaction::TransactionStatus = h.publish_package_with_options(
+        &existing_account,
+        &common::test_dir_path("test_option.data"),
+        build_options,
+    );
+    assert_success!(status);
+
+    h.enable_features(vec![FeatureFlag::ENABLE_ENUM_OPTION], vec![
+        FeatureFlag::NEW_OPTION_MODULE,
+    ]);
+    let status = h.run_entry_function(
+        &existing_account,
+        str::parse(
+            format!(
+                "0x{}::test_option::entry_function",
+                existing_account_addr.to_hex()
+            )
+            .as_str(),
+        )
+        .unwrap(),
+        vec![],
+        vec![MoveValue::U128(2).simple_serialize().unwrap()],
+    );
+    assert_success!(status);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_option_local() {
+    let mut h = MoveHarness::new();
+    h.enable_features(vec![], vec![
+        FeatureFlag::ENABLE_ENUM_OPTION,
+        FeatureFlag::NEW_OPTION_MODULE,
+    ]);
+
+    let existing_account =
+        h.new_account_at(AccountAddress::from_hex_literal(TESTNET_ACCOUNT_ADDR).unwrap());
+
+    let build_options = aptos_framework::BuildOptions {
+        bytecode_version: Some(8),
+        language_version: Some(LanguageVersion::V2_2),
+        ..Default::default()
+    };
+    let status: aptos_types::transaction::TransactionStatus = h.publish_package_with_options(
+        &existing_account,
+        &common::test_dir_path("test_option.data"),
+        build_options,
+    );
+    assert_success!(status);
+
+    h.enable_features(vec![FeatureFlag::ENABLE_ENUM_OPTION], vec![
+        FeatureFlag::NEW_OPTION_MODULE,
+    ]);
+    let status = h.run_entry_function(
+        &existing_account,
+        str::parse(
+            format!(
+                "0x{}::test_option::entry_function",
+                existing_account.address().to_hex()
+            )
+            .as_str(),
+        )
+        .unwrap(),
+        vec![],
+        vec![MoveValue::U128(2).simple_serialize().unwrap()],
+    );
+    assert_success!(status);
```

Comment:
Up until this point, this test is identical to one above? We can merge them together to save some CI time?

Reply:
The above one uses the old framework while this one uses the local new framework. Maybe we want to keep them separate for testing both cases?

Reply:
Missed that, sounds good 👍